src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java

```diff
@@ -646,15 +638,10 @@ private Mono<Tuple2<TransportEvent, TransportEvent>> fetchTransportEventByTransp
   private Mono<Vessel> fetchVesselByTransportCallId(String transportCallId) {
 
     if (transportCallId == null) return Mono.empty();
-    return transportCallRepository
-        .findById(transportCallId)
-        .flatMap(
-            x -> {
-              if (x.getVesselID() == null) {
-                return Mono.empty();
-              }
-              return vesselRepository.findById(x.getVesselID());
-            });
+    return fetchTransportCallById(transportCallId)
+        .filter(transportCall -> Objects.nonNull(transportCall.getVesselID()))
+        .switchIfEmpty(Mono.empty())
```

Comment:
`         .switchIfEmpty(Mono.empty())` is a no op. Did you mean to add something else here (like a Mono.error)?

Reply:
No, took a too literal way of translating the if statement to a filter. The switchIfEmpty is removed